### PR TITLE
Fix cf.insert crash when capacity < 0

### PR DESCRIPTION
**Pull request by liuchong**
When call CF.INSERT with a negative CAPACITY，it will crash the server:

Start server:

```
docker run --rm -p 6379:6379 -it redislabs/rebloom
```

Test commands:

```
127.0.0.1:6379> flushall
OK
127.0.0.1:6379> bf.insert x c -1 i iii
(error) Bad argument received
127.0.0.1:6379> flushall
OK
127.0.0.1:6379> cf.insert x c -1 i iii
Error: Server closed the connection
127.0.0.1:6379>
```

Error logs:

```
1:C 07 Apr 2022 05:44:26.678 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 07 Apr 2022 05:44:26.678 # Redis version=6.2.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 07 Apr 2022 05:44:26.678 # Configuration loaded
1:M 07 Apr 2022 05:44:26.679 * monotonic clock: POSIX clock_gettime
                _._
           _.-``__ ''-._
      _.-``    `.  `_.  ''-._           Redis 6.2.5 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 1
  `-._    `-._  `-./  _.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |           https://redis.io
  `-._    `-._`-.__.-'_.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |
  `-._    `-._`-.__.-'_.-'    _.-'
      `-._    `-.__.-'    _.-'
          `-._        _.-'
              `-.__.-'

1:M 07 Apr 2022 05:44:26.680 # Server initialized
1:M 07 Apr 2022 05:44:26.680 * Module 'bf' loaded from /usr/lib/redis/modules/redisbloom.so
1:M 07 Apr 2022 05:44:26.680 * Ready to accept connections


=== REDIS BUG REPORT START: Cut & paste starting from here ===
1:M 07 Apr 2022 05:44:29.658 # Redis 6.2.5 crashed by signal: 8, si_code: 1
1:M 07 Apr 2022 05:44:29.658 # Crashed running the instruction at: 0x7f9e4a5f6f14

------ STACK TRACE ------
EIP:
/usr/lib/redis/modules/redisbloom.so(+0x9f14)[0x7f9e4a5f6f14]

Backtrace:
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12980)[0x7f9e4b5e3980]
/usr/lib/redis/modules/redisbloom.so(+0x9f14)[0x7f9e4a5f6f14]
/usr/lib/redis/modules/redisbloom.so(+0xa017)[0x7f9e4a5f7017]
/usr/lib/redis/modules/redisbloom.so(CuckooFilter_Insert+0x4e)[0x7f9e4a5f754e]
/usr/lib/redis/modules/redisbloom.so(+0x4d7a)[0x7f9e4a5f1d7a]
/usr/lib/redis/modules/redisbloom.so(+0x4f7c)[0x7f9e4a5f1f7c]
redis-server *:6379(RedisModuleCommandDispatcher+0x66)[0x563d9bad9086]
redis-server *:6379(call+0x9c)[0x563d9ba4d84c]
redis-server *:6379(processCommand+0x66e)[0x563d9ba4f4de]
redis-server *:6379(processCommandAndResetClient+0x1c)[0x563d9ba62e4c]
redis-server *:6379(processInputBuffer+0xea)[0x563d9ba656ca]
redis-server *:6379(+0xfadea)[0x563d9bafadea]
redis-server *:6379(aeProcessEvents+0x1b7)[0x563d9ba45fe7]
redis-server *:6379(aeMain+0x1d)[0x563d9ba463dd]
redis-server *:6379(main+0x30a)[0x563d9ba4284a]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7)[0x7f9e4b201bf7]
redis-server *:6379(_start+0x2a)[0x563d9ba42d6a]

------ REGISTERS ------
1:M 07 Apr 2022 05:44:29.659 #
RAX:65b33f890432f104 RBX:65b33f890432f104
RCX:0000000000000001 RDX:0000000000000000
RDI:00007f9e4ac09a38 RSI:65b33f890432f104
RBP:00007f9e4ac7c850 RSP:00007fff6a7fb218
R8 :0000000000000002 R9 :0000000000000000
R10:65b33f8c657e5ebf R11:0000000000000017
R12:65b33f8c657e5ebf R13:00007f9e4ac7c860
R14:00007f9e4ac162f8 R15:00007f9e4ac7c860
RIP:00007f9e4a5f6f14 EFL:0000000000010246
CSGSFS:002b000000000033
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb227) -> 00007fff6a7fb340
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb226) -> ec8f7d88748dc200
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb225) -> 00007fff6a7fb30f
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb224) -> 65b33f8c657e5ebf
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb223) -> 65b33f890432f104
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb222) -> 00007f9e4a5f754e
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb221) -> 00007f9e4a7ff4c0
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb220) -> 00007f9e4ac0fb98
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb21f) -> 0000000000000001
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb21e) -> 00007fff6a7fb340
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb21d) -> 0000000000000000
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb21c) -> 00007fff6a7fb3a0
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb21b) -> 00007f9e4ac22200
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb21a) -> 00007fff6a7fb270
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb219) -> 0000000000000001
1:M 07 Apr 2022 05:44:29.659 # (00007fff6a7fb218) -> 00007f9e4a5f7017

------ INFO OUTPUT ------
# Server
redis_version:6.2.5
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:bf1309560f7fc5c
redis_mode:standalone
os:Linux 5.10.104-linuxkit x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:c11-builtin
gcc_version:7.5.0
process_id:1
process_supervised:no
run_id:0b516ca670c960392ce5297bcaa8f32210d43f9e
tcp_port:6379
server_time_usec:1649310269657926
uptime_in_seconds:3
uptime_in_days:0
hz:10
configured_hz:10
lru_clock:5143101
executable:/data/redis-server
config_file:
io_threads_active:0

# Clients
connected_clients:1
cluster_connections:0
maxclients:10000
client_recent_max_input_buffer:0
client_recent_max_output_buffer:0
blocked_clients:0
tracking_clients:0
clients_in_timeout_table:0

# Memory
used_memory:894112
used_memory_human:873.16K
used_memory_rss:7708672
used_memory_rss_human:7.35M
used_memory_peak:894112
used_memory_peak_human:873.16K
used_memory_peak_perc:105.12%
used_memory_overhead:829720
used_memory_startup:829648
used_memory_dataset:64392
used_memory_dataset_perc:99.89%
allocator_allocated:1048464
allocator_active:1208320
allocator_resident:3870720
total_system_memory:20992266240
total_system_memory_human:19.55G
used_memory_lua:37888
used_memory_lua_human:37.00K
used_memory_scripts:0
used_memory_scripts_human:0B
number_of_cached_scripts:0
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
allocator_frag_ratio:1.15
allocator_frag_bytes:159856
allocator_rss_ratio:3.20
allocator_rss_bytes:2662400
rss_overhead_ratio:1.99
rss_overhead_bytes:3837952
mem_fragmentation_ratio:9.29
mem_fragmentation_bytes:6879024
mem_not_counted_for_evict:0
mem_replication_backlog:0
mem_clients_slaves:0
mem_clients_normal:0
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0
lazyfreed_objects:0

# Persistence
loading:0
current_cow_size:0
current_cow_size_age:0
current_fork_perc:0.00
current_save_keys_processed:0
current_save_keys_total:0
rdb_changes_since_last_save:0
rdb_bgsave_in_progress:0
rdb_last_save_time:1649310266
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0
module_fork_in_progress:0
module_fork_last_cow_size:0

# Stats
total_connections_received:1
total_commands_processed:0
instantaneous_ops_per_sec:0
total_net_input_bytes:57
total_net_output_bytes:0
instantaneous_input_kbps:0.00
instantaneous_output_kbps:0.00
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
expired_stale_perc:0.00
expired_time_cap_reached_count:0
expire_cycle_cpu_milliseconds:0
evicted_keys:0
keyspace_hits:0
keyspace_misses:0
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
total_forks:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
tracking_total_keys:0
tracking_total_items:0
tracking_total_prefixes:0
unexpected_error_replies:0
total_error_replies:0
dump_payload_sanitizations:0
total_reads_processed:1
total_writes_processed:0
io_threaded_reads_processed:0
io_threaded_writes_processed:0

# Replication
role:master
connected_slaves:0
master_failover_state:no-failover
master_replid:118059789cddb87da63d82f5877010c29c8f12e7
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:0.023887
used_cpu_user:0.026276
used_cpu_sys_children:0.002616
used_cpu_user_children:0.000953
used_cpu_sys_main_thread:0.016556
used_cpu_user_main_thread:0.023178

# Modules
module:name=bf,ver=20209,api=1,filters=0,usedby=[],using=[],options=[]

# Commandstats

# Errorstats

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=1,expires=0,avg_ttl=0

------ CLIENT LIST OUTPUT ------
id=3 addr=172.17.0.1:63202 laddr=172.17.0.2:6379 fd=10 name= age=0 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=57 qbuf-free=40897 argv-mem=17 obl=4 oll=0 omem=0 tot-mem=61505 events=r cmd=cf.insert user=default redir=-1

------ CURRENT CLIENT INFO ------
id=3 addr=172.17.0.1:63202 laddr=172.17.0.2:6379 fd=10 name= age=0 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=57 qbuf-free=40897 argv-mem=17 obl=4 oll=0 omem=0 tot-mem=61505 events=r cmd=cf.insert user=default redir=-1
argv[0]: 'cf.insert'
argv[1]: 'x'
argv[2]: 'c'
argv[3]: '-1'
argv[4]: 'i'
argv[5]: 'iii'
1:M 07 Apr 2022 05:44:29.659 # key 'x' found in DB containing the following object:
1:M 07 Apr 2022 05:44:29.659 # Object type: 5
1:M 07 Apr 2022 05:44:29.659 # Object encoding: 0
1:M 07 Apr 2022 05:44:29.659 # Object refcount: 1

------ MODULES INFO OUTPUT ------

------ FAST MEMORY TEST ------
1:M 07 Apr 2022 05:44:29.660 # Bio thread for job type #0 terminated
1:M 07 Apr 2022 05:44:29.660 # Bio thread for job type #1 terminated
1:M 07 Apr 2022 05:44:29.660 # Bio thread for job type #2 terminated
*** Preparing to test memory region 563d9be10000 (2281472 bytes)
*** Preparing to test memory region 563d9d393000 (270336 bytes)
*** Preparing to test memory region 7f9e485ea000 (8388608 bytes)
*** Preparing to test memory region 7f9e48deb000 (8388608 bytes)
*** Preparing to test memory region 7f9e495ec000 (8388608 bytes)
*** Preparing to test memory region 7f9e49ded000 (8388608 bytes)
*** Preparing to test memory region 7f9e4a800000 (8388608 bytes)
*** Preparing to test memory region 7f9e4b5cd000 (16384 bytes)
*** Preparing to test memory region 7f9e4b7ec000 (16384 bytes)
*** Preparing to test memory region 7f9e4bcb8000 (12288 bytes)
*** Preparing to test memory region 7f9e4c911000 (32768 bytes)
*** Preparing to test memory region 7f9e4c91d000 (4096 bytes)
.O.O.O.O.O.O.O.O.O.O.O.O
Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.

------ DUMPING CODE AROUND EIP ------
Symbol: (null) (base: (nil))
Module: /usr/lib/redis/modules/redisbloom.so (base 0x7f9e4a5ed000)
$ xxd -r -p /tmp/dump.hex /tmp/dump.bin
$ objdump --adjust-vma=(nil) -D -b binary -m i386:x86-64 /tmp/dump.bin
------

=== REDIS BUG REPORT END. Make sure to include from START to END. ===

       Please report the crash by opening an issue on github:

           http://github.com/redis/redis/issues

  Suspect RAM error? Use redis-server --test-memory to verify it.
```

### CODE REVIEW

**Comment by liuchong** [comment]
### Test after fix

* Server
   ```
   redis-server --loadmodule ./redisbloom.so
   ```
* Client
   ```
   127.0.0.1:6379> flushall
   OK
   127.0.0.1:6379> bf.insert x c -1 i iii
   (error) Bad argument received
   127.0.0.1:6379> flushall
   OK
   127.0.0.1:6379> cf.insert x c -1 i iii
   (error) Bad argument received
   127.0.0.1:6379>
   ```

**Comment by liuchong** [comment]
Branch is rebase updated, please check @ashtul 👀

**Comment by chayim** [comment]
@liuchong Thanks for the great fix!

**Review by chayim — DISMISSED** [review]

**Comment by liuchong** [comment]
`cla` have already signed, not sure why it is still showing a `license/cla Expected — Waiting for status to be reported` 🤔

**Comment by liuchong** [comment]
`cla` have already signed, not sure why it is still showing a `license/cla Expected — Waiting for status to be reported` 🤔  @chayim

**Comment by liuchong** [comment]
Is there anything needs to change? @chayim @ashtul 👀

**Comment by liuchong** [comment]
I rebased against master branch with no change, please check @chayim @ashtul 👀

File: src/rebloom.c [hunk 1]
```diff
@@ -676,6 +678,11 @@ static int CFInsert_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv,
     if (items_pos < 0 || items_pos == argc) {
         return RedisModule_WrongArity(ctx);
     }
+
+    if (options.capacity < 1) {
```
Comment (ashtul) [inline src/rebloom.c:682]: Please change to `2 * CF_DEFAULT_BUCKETSIZE`, similar to the limit in `CFReserve_RedisCommand` and move into the switch at the end of the case `c`.

**Review by ashtul — COMMENTED** [review]
Thanks for the contribution.
Can you please add a test?

File: src/rebloom.c (hunk 1 above)
Comment (liuchong) [reply src/rebloom.c]: Done @ashtul  👀

**Comment by liuchong** [comment]
Resolved the review conversation and also added test for CF.INSERT and CF.INSERTNX with bad capacity, please check again @ashtul 👀

**Review by ashtul — APPROVED** [review]
Thanks for adding the tests.
Approved

**Pull request merged**